components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/TenantMgtImpl.java

```diff
@@ -0,0 +1,355 @@
+import org.wso2.carbon.stratos.common.util.CommonUtil;
+import org.wso2.carbon.tenant.mgt.core.TenantPersistor;
+import org.wso2.carbon.tenant.mgt.core.exception.TenantManagementClientException;
+import org.wso2.carbon.tenant.mgt.core.exception.TenantManagementServerException;
+import org.wso2.carbon.tenant.mgt.core.exception.TenantMgtException;
+import org.wso2.carbon.tenant.mgt.core.internal.TenantMgtCoreServiceComponent;
+import org.wso2.carbon.tenant.mgt.internal.TenantMgtServiceComponent;
+import org.wso2.carbon.tenant.mgt.util.TenantMgtUtil;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.api.TenantMgtConfiguration;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.config.multitenancy.MultiTenantRealmConfigBuilder;
+import org.wso2.carbon.user.core.constants.UserCoreClaimConstants;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.tenant.Tenant;
+import org.wso2.carbon.user.core.tenant.TenantManager;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.wso2.carbon.tenant.mgt.util.TenantMgtUtil.initializeTenantInfoBean;
+
+/**
+ * Default implementation of {@link org.wso2.carbon.tenant.mgt.TenantMgtService} interface.
+ */
+public class TenantMgtImpl implements TenantMgtService {
+
+    private static final Log log = LogFactory.getLog(TenantMgtImpl.class);
+    private static final String ADD_ADMIN_TRUE = "true";
+
+    /**
+     * super admin adds a tenant.
+     *
+     * @param tenant tenant information.
+     * @return UUID uuid used to represent the tenant.
+     * @throws TenantMgtException if error in adding new tenant.
+     */
+    public String addTenant(Tenant tenant) throws TenantMgtException {
+
+        try {
+            CommonUtil.validateEmail(tenant.getEmail());
+        } catch (Exception e) {
+            if (e instanceof TenantClientException) {
+                throw new TenantManagementClientException(((TenantClientException) e).getErrorCode(), e.getMessage());
+            } else if (e instanceof TenantServerException) {
+                throw new TenantManagementServerException(e.getMessage(), e);
+            }
+        }
+
+        String tenantDomain = tenant.getDomain();
+        String resourceId;
+        int tenantId;
+        TenantPersistor persistor = new TenantPersistor();
+        try {
+            TenantMgtUtil.validateDomain(tenantDomain);
+            checkIsSuperTenantInvoking();
+
+            // Set a thread local variable to identify the operations triggered for a tenant admin user.
+            TenantMgtUtil.setTenantAdminCreationOperation(true);
+
```

Comment:
Create private methods for input validation, tenant creation, admin creation and call it here.